File: snapshot/git_test.swift
import Foundation

func generateRandomNumbers() -> [Int] {
    var randomNumbers: [Int] = []
    for _ in 1...10 {
        let randomNumber = Int.random(in: 1...100)
        randomNumbers.append(randomNumber)
    }
    return randomNumbers
}

func isPrime(_ number: Int) -> Bool {
    if number < 2 { return false }
    for i in 2...Int(sqrt(Double(number))) {
        if number % i == 0 {
            return false
        }
    }
    return true
}

let randomNumbers = generateRandomNumbers()

print("Random Numbers:")
print(randomNumbers)

let primeNumbers = randomNumbers.filter { isPrime($0) }
let averagePrime = primeNumbers.isEmpty ? 0 : Double(primeNumbers.reduce(0, +)) / Double(primeNumbers.count)

print("Prime Numbers:")
print(primeNumbers)
print("Average of Prime Numbers: \(averagePrime)")

let originalString = "Hello, World!"
let reversedString = String(originalString.reversed())

print("Original String: \(originalString)")
print("Reversed String: \(reversedString)")

class Dog {
    let name: String
    let breed: String
    
    init(name: String, breed: String) {
        self.name = name
        self.breed = breed
    }
    
    func bark() {
        print("\(name) is barking!")
    }
}

let myDog = Dog(name: "Buddy", breed: "Golden Retriever")
myDog.bark()

File: snapshot/git_test.py
import random

# Create a list of random numbers
random_numbers = [random.randint(1, 100) for _ in range(10)]

# Function to check if a number is prime
def is_prime(num):
    if num < 2:
        return False
    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False
    return True

# Filter prime numbers from the list
prime_numbers = [num for num in random_numbers if is_prime(num)]

# Calculate the average of the prime numbers
if prime_numbers:
    average_prime = sum(prime_numbers) / len(prime_numbers)
else:
    average_prime = 0

# Print the original list
print("Random Numbers:")
print(random_numbers)

# Print the prime numbers
print("Prime Numbers:")
print(prime_numbers)

# Print the average of prime numbers
print("Average of Prime Numbers:", average_prime)

# Function to reverse a string
def reverse_string(input_str):
    return input_str[::-1]

# Example usage
original_string = "Hello, World!"
reversed_string = reverse_string(original_string)

# Print the reversed string
print("Original String:", original_string)
print("Reversed String:", reversed_string)

# Class definition
class Dog:
    def __init__(self, name, breed):
        self.name = name
        self.breed = breed

    def bark(self):
        print(f"{self.name} is barking!")

# Create a Dog instance
my_dog = Dog("Buddy", "Golden Retriever")

# Call the bark method
my_dog.bark()
